src/argparse.lua

```diff
@@ -1087,6 +1043,46 @@ function Parser:get_help()
    return table.concat(blocks, "\n\n")
 end
 
+function Parser:add_help_command(value)
+   if value then
+      assert(type(value) == "string" or type(value) == "table",
+         ("bad argument #1 to 'add_help_command' (string or table expected, got %s)"):format(type(value)))
+   end
+
+   local help = self:command()
+      :description "Show help for commands."
+   help:argument "command"
+      :description "The command to show help for."
+      :args "?"
+      :action(function(_, _, cmd)
+         if not cmd then
+            print(self:get_help())
+            os.exit(0)
+         else
+            for _, command in ipairs(self._commands) do
+               for _, alias in ipairs(command._aliases) do
+                  if alias == cmd then
+                     print(command:get_help())
+                     os.exit(0)
+                  end
+               end
+            end
+         end
+         help:error(("unknown command '%s'"):format(cmd))
+      end)
+
+   if value then
+      help = help(value)
+   end
+
+   if not help._name then
+      help "help"
+   end
+
+   self._help_command_idx = #self._commands
```

Comment:
`_help_command_idx` seems to be unused now

Reply:
It was used for completions but it would be easier to save the command instead of the index. I'll do that in the completion PR.

Reply:
It was used for completions but it would be easier to save the command instead of the index. I'll do that in the completion PR.